File: leetcode/1207. Unique Number of Occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> count;
        for (const int& num : arr) {
            count[num] += 1;
        }

        unordered_set<int> s;
        for (const auto& [num, c] : count) {
            if (s.find(c) != s.end()) {
                return false;
            }
            s.insert(c);
        }
        return true;
    }
};

File: leetcode/1207. Unique Number of Occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> count = new HashMap<>();
        for(int num : arr){
            count.putIfAbsent(num, 0);
            count.put(num, count.get(num) + 1);
        }

        boolean[] occurred = new boolean[1001];
        for(Map.Entry<Integer, Integer> e : count.entrySet()){
            if(occurred[e.getValue()]){
                return false;
            }
            occurred[e.getValue()] = true;
        }
        return true;
    }
}